great_expectations/gx/plugins/expectations/expect_column_values_to_have_list_length.py

```diff
@@ -0,0 +1,142 @@
+import pandas as pd
+from typing import Optional, Any
+
+from great_expectations.core.expectation_configuration import ExpectationConfiguration
+from great_expectations.execution_engine import PandasExecutionEngine
+from great_expectations.expectations.expectation import ColumnMapExpectation
+from great_expectations.expectations.metrics import (
+    ColumnMapMetricProvider,
+    column_condition_partial,
+)
+
+
+# This class defines a Metric to support your Expectation.
+# For most ColumnMapExpectations, the main business logic for calculation will live in this class.
+class ColumnValuesListLength(ColumnMapMetricProvider):
+    """Class definition for list length checking metric."""
+
+    # This is the id string that will be used to reference your metric.
+    condition_metric_name = "column_values.list_length"
+    condition_value_keys = ("list_length",)
+
+    # This method implements the core logic for the PandasExecutionEngine
+    @column_condition_partial(engine=PandasExecutionEngine)
+    def _pandas(cls, column: pd.core.series.Series, list_length: int, **kwargs) -> bool:
+        """Core logic for list length checking metric on a
+        pandas execution engine.
+
+        Args:
+            column (pd.core.series.Series): Pandas column to be evaluated.
+            list_length (int): Expected list length.
+        Returns:
+            bool: Whether or not the column values have the expected list length.
+        """
+        return column.apply(lambda x: cls._check_list_length(x, list_length))
+
+    @staticmethod
+    def _check_list_length(cell: Any, list_length: int) -> bool:
+        """Check if a cell is a list, and if it has the expected length.
+
+        Args:
+            cell (Any): Individual cell to be evaluated.
+            list_length (int): Expected list length.
+
+        Returns:
+            bool: Whether or not the cell is a list with the expected length.
+        """
+        if not isinstance(cell, list):
+            return False
+        if len(cell) != list_length:
+            return False
+        return True
+
+    # This method defines the business logic for evaluating your metric when using a SqlAlchemyExecutionEngine
+    # @column_condition_partial(engine=SqlAlchemyExecutionEngine)
```

Comment:
nit: Can probably remove this commented out code

Reply:
I was originally going to leave this in case we wanted to support different execution engines for these expectations in the future (for example, if they are used in other projects), but I suppose we can just refer to the original [template](https://github.com/great-expectations/great_expectations/blob/develop/examples/expectations/column_map_expectation_template.py) if we do decide to do that.